### DISCUSSION TITLE
Uncaught exception: No training images were found for NeRF training!

### DISCUSSION THREAD

**Discussion by Han437429** [NVlabs/instant-ngp#534] [Q&A]
INFO     Loading NeRF dataset from
INFO       data/nerf/stuffed/transforms.json
ERROR    Uncaught exception: No training images were found for NeRF training!

**Comment by Tom94** [comment]
This means that the paths in your transforms.json don't point to images. To make sure: they have to be relative to the location of transforms.json.

**Comment by YokuTanaka** [comment]
I also got an error in a similar area.
I racked my brains for a long time, but for some reason the fox was successful, so I tried changing the file path of the image in "transforms" to the same path as the fox, and it worked.
I would like to find a solution to this problem, because it is quite difficult to change the paths of a large number of images every time.
Sorry for the machine translation...

**Reply by gvbgeomatics** [reply]
able to fix it and configured the same on cloud successfully. We are able to automate the deployment process of Instant Nerf on cloud so the developers and content developers does not have to spend hours of time in the setup rather focus on content development and 3d modeling. More details here: https://nebulacloud.ai/connect/blogs/how-to-use-nvidia-instant-nerf-on-nebula-cloud

**Reply by jaharoni** [reply]
Hi there! I was just checking out NebulaCloud for this reason, have the same error after a week of compiling. I'd rather not go through it again if there's a way around it, if you have some insights to share. nerf noob here

**Reply by gvbgeomatics** [reply]
Hi there, No need to compile anything on Nebula Cloud (https://nebulacloud.ai). Everything comes preconfigured and is ready to use. Just prepare your dataset and run the testbed directly. User Guide provided inside on how to use. Rest all can be run directly on Nebula Cloud. For technical queries and support, please write to support@nebulacloud.in. Happy to assist you!

For more details on how to configure your dataset, refer this link:https://github.com/NVlabs/instant-ngp/blob/master/docs/nerf_dataset_tips.md

Few samples of NerF Scenes created on Nebula Cloud:
https://www.youtube.com/watch?v=SHUVlqiOyiU
https://www.youtube.com/watch?v=zy-8KZaWa-w
https://www.youtube.com/watch?v=4jJZR7BSZIo

**Reply by jaharoni** [reply]
Amazing, thank you!

To be clear though, the process that just ran for a week and gave me this error would have to be recreated yes?:
FileNotFoundError: [Errno 2] No such file or directory: 'data/Maine/transforms.json'

I'm very new at this and could be conflating two steps here. Basically I have a folder with over 3k images that were created by the compiler super quick but it was (I think) the building of the dataset that couldn't be matched with a transforms.json file. Idk

I'll head to the site and start messing around for sure!

**Comment by PascalHbr** [comment]
I also face this issue. I think it would be sensible to adjust the relative location to the `transforms.json` file based on the given `--out` flag. For example. I have my images in `data/nerf/example_images/images` and also my `--out` flag to `data/nerf/example_images`. However, the paths in the `transforms.json` file point to `data/nerf/example_images/images` instead of `images`.

**Reply by gvbgeomatics** [reply]
Use Instant NerF without any compilation directly on Cloud. Refer this link: https://nebulacloud.ai/connect/blogs/how-to-use-nvidia-instant-nerf-on-nebula-cloud

**Reply by yoponchik** [reply]
@PascalHbr @gvbgeomatics Is there a way to do this without doing it on the cloud? I also got the `transforms.json` in a different folder using the `--out` parameter, but I'm getting the same error.

**Reply by gvbgeomatics** [reply]
Could you please share the complete log here?

**Comment by gvbgeomatics** [comment]
Some more exciting updates from Nebula Cloud: We have launched cloud based workbench for Neural Radiance Fields (NeRF) where all the NeRF implementations are preconfigured and ready to use with the choice of GPU and storage configurations. It comes with latest build of Instant NGP, NeRF Studio Latest Version 1.0, Postshot Version 3.0 - Beta along with popular photogrammetry softwares like Agisoft Metashape, Reality Capture, Blender and many more, all at one in place. 

More details here: https://nebulacloud.ai/connect/blogs/how-to-use-gaussian-splatting-with-nerf-studio-on-nebula-cloud
